target/linux/kirkwood/base-files/lib/preinit/07_set_iface_mac

```diff
@@ -19,6 +19,14 @@ preinit_set_mac_address() {
 		ip link set dev dmz address $(macaddr_add $mac 8) 2>/dev/null
 		ip link set dev dsl address $(macaddr_add $mac 9) 2>/dev/null
 		;;
+	endian,4i-edge-200)
+		mac=$(cat /sys/class/net/eth0/address)
```

Comment:
No "direct" source available (e.g. from flash)?

Reply:
It's problem:

```
root@OpenWrt:/# hexdump /dev/mtd1 > /dev/null
[  429.067078] __nand_correct_data: uncorrectable ECC error
[  429.072420] __nand_correct_data: uncorrectable ECC error
[  429.078960] __nand_correct_data: uncorrectable ECC error
[  429.084340] __nand_correct_data: uncorrectable ECC error
[  429.091065] __nand_correct_data: uncorrectable ECC error
[  429.096426] __nand_correct_data: uncorrectable ECC error
[  429.101753] __nand_correct_data: uncorrectable ECC error
[  429.108625] __nand_correct_data: uncorrectable ECC error
[multiple the same errors]
```

Stock Marvell u-boot uses his own nand ecc method for u-boot end env partition. When I save env in OpenWrt, it was unreadable from u-boot. I can't modify it and reading causes trash in dmesg.  That's the reason why I choose sysfs.

Reply:
Ah, that explains.
BTW then shouldn't we set u-boot and u-boot-env partitions as read-only?

Reply:
I marked it as read-only.